zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/immutable/DecisionState.java

```diff
@@ -92,6 +93,38 @@ Optional<DeployedDrg> findDecisionRequirementsByTenantAndKey(
   List<PersistedDecision> findDecisionsByTenantAndDecisionRequirementsKey(
       final String tenantId, long decisionRequirementsKey);
 
+  /**
+   * Iterates over all persisted decision requirements until the visitor returns false or all
+   * decision requirements have been visited. If {@code previousDecisionRequirements} is not null,
+   * the iteration skips all decision requirements that appear before it. The visitor is
+   * <em>not</em> called with a copy of the decision requirements to avoid needless copies of the
+   * relatively large {@link PersistedDecisionRequirements} instances.
+   */
+  void forEachDecisionRequirements(
+      final DecisionRequirementsIdentifier previousDecisionsRequirements,
+      final PersistedDecisionRequirementsVisitor visitor);
+
+  /**
+   * Iterates over all persisted decisions until the visitor returns false or all decisions have
+   * been visited. If {@code previousDecision} is not null, the iteration skips all decisions that
+   * appear before it. The visitor is <em>not</em> called with a copy of the decision to avoid
+   * needless copies of the relatively large {@link PersistedDecision} instances.
+   */
+  void forEachDecision(
+      final DecisionIdentifier previousDecision, final PersistedDecisionVisitor visitor);
```

Comment:
🔧 I guess you don't even need this method anymore now that you always iterate over the DRGs